### ISSUE TITLE
How to Use bash aliases inside cmd bash -c command?

### ISSUE THREAD

**Issue by g0ld3lux** [Microsoft/WSL#1437]
I know i can do from cmd
```
 bash -c  "composer $*"
```
but what if i have a bash functions that i wanted to be used in cmd?

lets say i have this function
```
function routes()
{
 if [$# -eq 0 ]; then
         php artisan route:list
else
        php artisan route:list | grep ${1}
fi
}
```
if i do try to run

bash -c "routes api"

i get this error
/bin/bash: routes : command not found

How can i Use my aliases? inside this command
bash -c ???

Can SOmeone point me to the right direction thanks

**Comment by g0ld3lux** [comment]
so as a work around i tried to 
do something like this
first i tried

bash -c "php artisan route:list | ack api"

it work  and got the result i wanted

```
|        | GET|HEAD | api/user               |          | Closure                                                                | api,auth:api |
```

so i have added then to my cmdrc

this
```
DOSKEY rf=bash -c "php artisan route:list | ack $*"
```

So is there any a way to work around this more fluently by using my bashrc aliases?

if someone know how to do it please advice me thanks

**Comment by FrostWalker** [comment]
Your function is established in .bashrc correct? When you call bash from windows you need to use the -i flag to launch an interactive bash shell and read .bashrc, otherwise it will be ignored.

So what you want to do is run `bash -ic "routes api"` from cmd instead

**Comment by rodrymbo** [comment]
If I use bash in Linux, it usually results in a "login" shell, which runs both .bash_profile and .bashrc (after the corresponding system files) .

On my WSL, if I run `bash.exe` (or `bash.exe -i`) it only runs `.bash_profile`. If I run `bash.exe -l` I get the usual login shell, which runs both ~/.bash_profile and ~/.bashrc.  (I've tweaked things on my machine, so there's a possibility my setup is unusual, but it shouldn't take much testing to see whether your setup works the same.)

You can also get to a "login" shell by using `bash -l`, from the bash interactive shell if you don't want to go out to CMD and back in again. Or you get a login shell if you go in through ssh. 

In other words, if you are used to bash on Linux giving you a login shell, you might be surprised unless you use the -l parameter on WSL.

**Comment by FrostWalker** [comment]
@rodrymbo `-i` should launch a bash with .bashrc, in my tests a function set in .bashrc works by calling `bash -ic sometestfunction` but fails with command not found using just `bash -c sometestfunction`. Additionally, running bash.exe directly with no arguments should definitely read .bashrc, there must be something screwy in your set-up if it doesn't.

**Comment by g0ld3lux** [comment]
My initial set up was this...
non root user as the default user that uses zsh instead of bash...

so what your saying is false at that kind of set up...

so what i did was ...

make root as the default user , which is using bash...

and since my bashrc and zshrc is sourcing the same bash_profile/aliases

changing shell  fixes the issue

and that fixes the issue ...

when i run bash -ic routes api

so now i can  use 
 ```
alias routes=bash -ic "routes $*"

```

**Comment by FrostWalker** [comment]
@g0ld3lux using zsh complicates things. WSL doesn't respect a user's default shell, and will just run /bin/bash automatically. How is your non-root default user set up to run zsh instead of bash?

**Comment by g0ld3lux** [comment]
I finally set up my Windows Dev Machine for laravel the way i wanted to...
i created a gist (guide) how to achieve what i have in my own laptop...
https://gist.github.com/g0ld3lux/db5c2aa472d7815728a6990abc599df2#file-readme-md

**Comment by therealkenc** [comment]
Discussion ran course.
